### ISSUE TITLE
Building Materials In Unity3d Core Doesn't Give Correct Result, Yet HDRP

### ISSUE THREAD

**Issue by DireLykaios** [soupday/cc_unity_tools_3D#39]
Hi!
I'm having issue building materials in Unity3d Core, it gives me a messed up result. I kinda need it to properly build materials in this version because I will be using other tools that only work in Core.

As you can see, everything works well in blender after exporting from CC4. My issue happens when I export the fbx file to Unity and try to build the materials using the plugin. I've tested the material building in HDRP and it works perfectly, is there any way I can achieve the same result in Core?

Photos Attached
![Blender Works](https://user-images.githubusercontent.com/120895863/208284795-f0ce69c0-5e7b-42fc-9c5d-42e079454013.PNG)
![Unity Model Broken](https://user-images.githubusercontent.com/120895863/208284797-4c2176f3-c113-4b84-ab72-49338e14a86a.PNG)

**Comment by soupday** [comment]
it looks washed out, is your project set to linear color space? (Unity 3D core defaults to gamma color space, URP and HDRP default to linear)

![cs](https://user-images.githubusercontent.com/79094830/208291919-7a865a15-80fd-4331-8dc0-ca17709e8823.jpg)

**Comment by DireLykaios** [comment]
Hi, it is set to linear, and I also have the processing stack installed.
It’s like everything’s grey and metallic like.

**Comment by DireLykaios** [comment]
Do I need to rebuild the materials after changing to linear?

As I had mentioned everything works perfectly in HDRP so I know it’s an
issue with the material setup or way it’s displayed.

**Comment by soupday** [comment]
Is it just because you're in an new empty scene? The default empty scene lighting isn't very interesting (and there's no post processing set up in it). What does it look like in the tool's own preview scene?

**Comment by DireLykaios** [comment]
I messed around with the lighting, it doesn't seem to make much of a difference. For comparison here is a similarly textured character I had run through some other plugins. Something's going wonky with the material building on my end. 

![vrm conversion](https://user-images.githubusercontent.com/120895863/208329078-bc51af9c-a88c-44ca-91aa-8b708ddc6593.PNG)

**Comment by soupday** [comment]
Ok, I have a feeling I know what it is.

Coming from Blender, the textures aren't in a suitable format for Unity so they have to be baked: diffuse + alpha, metallic + roughness to metallic alpha smoothness, that sort of thing. But it only does this once, because it takes quite a long time, and then it re-uses that initial bake on subsequent rebuilds.

But... if the character is first build when the project is in Gamma color space, that gamma space is baked into the textures and they look washed out and too smooth, even when the project is then changed to Linear color space, e.g:

![washy](https://user-images.githubusercontent.com/79094830/208399865-5c64717d-4adc-4378-a18f-6554311581c6.jpg)

So, to fix it: There is a setting to rebake the Blender textures, turn it on and rebuild the character, that should fix it.

![settings](https://user-images.githubusercontent.com/79094830/208400596-edadfa50-2cab-4832-817a-764f14d6d839.jpg)

**Comment by DireLykaios** [comment]
Yep, that option and rebuilding the materials fixed the weird skin coloration and everything. Final question, if you do know, is how would I improve the hair transparency issues? 

Really appreciate your help so far, and I'd recommend documenting this troubleshooting process for others who may have had similar problems.
![Better](https://user-images.githubusercontent.com/120895863/208507414-5e759f4d-5331-46d1-9329-18171f8a86bc.PNG)

**Comment by DireLykaios** [comment]
I don't know if the hair just looks bad here, but when you transfer it to another program it renders better, or if there is something I can do with the way the hair is displayed in Unity at this step to resolve the way its rendering due to transparency issues.

**Comment by DireLykaios** [comment]
For example, here is the same hair, the right one was just imported through multiple other steps. Is there a way to achieve better hair rendering through your plugin, maybe with some additional steps?

![Hair](https://user-images.githubusercontent.com/120895863/208509573-a345f2b4-e4fe-4530-ac03-b9c3cb71c48c.PNG)

**Comment by soupday** [comment]
Use the prefab generated in the Prefabs folder. The Fbx can't be set up properly as all the mesh objects in it are read only, so the character is built in the prefabs.

The hair materials for two pass hair in particular don't work on the Fbx.

**Comment by soupday** [comment]
Alternatively there is the single pass hair, which is drawn as dithered opaque, faster performance, but it's the lowest quality. Or there is the MSAA Coverage Single pass hair shader which is also rendered opaque but does some kind of trickery with the MSAA anti-aliasing to make it look alpha blended. It's probably the fastest rendering performance but requires MSAA enabled, which may overall be slower. (And only seems to work in the Game window)

**Comment by DireLykaios** [comment]
Awesome, thanks. I've determined that the main issue I'm facing is that my hair material is rendering as a standard shader, with cutout as it's rendering mode. Putting the rendering mode as fade makes it more appealing for now, with some a transparency effect. 

I appreciate your help, and I'll try to build a better more authentic hair material for the avatar.

**Comment by soupday** [comment]
If you change the hair materials from Pbr to Digital Human Hair in CC4 before exporting, this will tell the Unity setup to treat it as hair and use the correct hair shaders.

It may need some tweaking, but it should improve things.

**Comment by DireLykaios** [comment]
We're in business now, baby. 
The hair became an issue where the normal maps were not being imported and created like other materials. I did it by hand for now. 

Also, if Unity is giving you problems the option “Blendshape Normals” to “None” or enable “Legacy Blendshape Normals”. That fixed some of my other shadow/mesh issues later on down the line.

![Fixing](https://user-images.githubusercontent.com/120895863/208775024-4b314db6-dfe3-4f3b-bb68-f79f14b2fb0e.PNG)
